### DISCUSSION TITLE
Navigating the map with the arrow keys

### DISCUSSION THREAD

**Discussion by autoca** [freeplane/freeplane#625] [Help]
I think this is huger than it may sound at first.
![grafik](https://user-images.githubusercontent.com/37442404/188198161-c479f018-c39d-4897-9a67-66433621a77f.png)

Explaining it with the image:
When I have node 1-1-1 selected and I press the down arrow, then node 1-2 gets selected.
So far, so good.
Now that 1-2 is selected, how do I go back up to 1-1-1 with pressing just ONE arrow key?
Obviously this is not possible. Pressing up arrow goes to 1-1 and thas is ok. Pressing the right arrow, however, does nothing but moving the map. This is soooooooooooooooooooooooo annoying.

I rather need for the down arrow to do nothing when there is no other node on the same level than having to jump through hoops in order to go back to 1-1-1. Because it happens way to often that I accidently go down like that.

I just figured out one work around: Ctrl+Alt+Left
This goes to the "previous node". I will try it but I don't find it very intuitive yet. Let's see.

Any experiences and feedback welcome. Thx.

Regards, autoca

**Comment by macmarrum** [comment]
> I rather need for the down arrow to do nothing when there is no other node on the same level than having to jump through loops in order to go back to 1-1-1.

I think a script can do the trick.

The script would check if the next visible branch has at least the same number of nodes. If yes, it would execute `FreeSelectNodeAction.DOWN`, otherwise no action.

You can assign `DOWN` hot key to the script, overriding the default one.

If you're starting your journey with Freeplane scripting, this might be a tricky task. Anyway, feel free to come back for help.

**Reply by autoca** [reply]
Thank you for your comfort and support!!

**Reply by Harliff** [reply]
Hello! I've faced this problem too. 
What is the best way to avoid jumping away from "1-1-1" node on press Up/Down key?
I'm newbie to the freeplane. Can you give me basic advice how to write required script?
